lib/realtime/vehicles.ex

```diff
@@ -57,6 +72,22 @@ defmodule Realtime.Vehicles do
     end)
   end
 
+  @spec sort_incoming_vehicles_and_ghosts(Route.id(), [Vehicle.t() | Ghost.t()], [Trip.t()]) :: [
+          Vehicle.t() | Ghost.t()
+        ]
+  defp sort_incoming_vehicles_and_ghosts(route_id, vehicles_and_ghosts, incoming_trips) do
+    vehicles_and_ghosts
+    |> Enum.sort_by(fn vehicle_or_ghost ->
+      incoming_trip =
+        incoming_trips
+        |> Enum.find(fn trip ->
+          trip.block_id == vehicle_or_ghost.block_id && trip.route_id == route_id
```

Comment:
What if a block is `[route 71, route 73, route 71]`, and a bus on the 2nd trip incoming into the 3rd trip. This could select the 1st trip instead of the 3rd, and give the wrong start time.

Reply:
Conceptually it would be simple to further filter trips by time, rejecting those that started in the past. I've been thinking about it though and here's a problem: our concept of "now" here is a Unix timestamp. The trip start/end times are given as GTFS times-of-day. There are multiple representations of any given timestamp in time-of-day format, to disambiguate you need a service date.

What I'm thinking about doing this is assuming that if the current wall-clock time is between midnight and, say, 5 AM, the service date is calendar yesterday. Otherwise it's calendar today.

Do you see anything wrong with that approach?

Reply:
Ah, I'm thinking of subway service, where there is a definite period every day without any service at all.